### PR TITLE
docs: document mount adapter requirements

### PR DESCRIPTION
**Pull request by lmiller1990**
For CT GA, we'd like to have a rough guide on what constitutes a Mount Adapter, and some guidelines to making new ones. This is useful both internally for future first party adapters, and third parties.. This is a first pass at documenting it.

### CODE REVIEW

File: npm/mount-utils/README.md [hunk 1]
```diff
@@ -8,6 +8,140 @@ It is used in:
...
+
+All Component Tests require a component to be **mounted**. This is generally done with a custom command, `cy.mount` by default.
+
+
+### Requirements
+
+All the functionality used to create the first party Mount adapters is available to support third parties adapters. At minimum, a Mount Adapter must:
+
+- Receive a component as the first argument. This could be class, function etc - depends on the framework.
+- Return a Cypress Chainable (for example using `cy.wrap`) that resolves whatever is idiomatic for your framework
```
Comment (ZachJW34) [inline npm/mount-utils/README.md:24]: I would add:
```
- Teardown the last mounted component between tests
```

and update the example to show the teardown.

File: npm/mount-utils/README.md [hunk 2]
```diff
@@ -8,6 +8,140 @@ It is used in:
...
+    .replace(/([a-z0–9])([A-Z])/g, "$1-$2")
+    .toLowerCase();
+
+  /// Register Web Component
+  maybeRegisterComponent(name, webComponent);
+
+  // Render HTML containing component.
+  $root.innerHTML = `<${name} id="root"></${name}>`;
+
+  injectStylesBeforeElement(options ?? {}, document, getContainerEl())
```
Comment (ZachJW34) [inline npm/mount-utils/README.md]: Docs above say to call this before mounting the component but this example calls it after.

File: npm/mount-utils/README.md (hunk 1 above)
Comment (lmiller1990) [reply npm/mount-utils/README.md:24]: Added your point - good idea.

I have [this line](https://github.com/cypress-io/cypress/pull/23976/files#diff-e5219acad52acdefe7159f0202d3fac79431644def4dcc106a3f70db4b8be0deR54)

```js
Cypress.on("test:before:run", () => {
    // Do some cleanup from previous test
  });
```

There actually isn't any teardown to be done here. The one thing I'd *like* to do is unregister the Web Component - after a solid google, though, this is currently unsupported! :(

Any other ideas/recommendations we could add for a cleanup example?

File: npm/mount-utils/README.md (hunk 2 above)
Comment (lmiller1990) [reply npm/mount-utils/README.md]: Nice catch, moved it.

Comment (ZachJW34) [reply npm/mount-utils/README.md]: Would it be enough to call `getContainerEl().innerHTML = ''`? Just to clear the element out before each run

Comment (lmiller1990) [reply npm/mount-utils/README.md]: Sure, sounds good.

Comment (lmiller1990) [reply npm/mount-utils/README.md]: done!

**Review by ZachJW34 — APPROVED** [review]

**Review by ZachJW34 — APPROVED** [review]
Nice work!!

**Review by rockindahizzy — APPROVED** [review]
Nice write up!

**Pull request merged**